### PR TITLE
py-cleo: add versions 2.0.0 2.0.1; add maintainers

### PR DESCRIPTION
**Pull request by LydDeb**

### CODE REVIEW

**Review by tldahlgren — COMMENTED** [review]
Confirmed the affected version sha256.  Deferring merge decision to @adamjstewart .

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 1]
```diff
@@ -12,17 +12,24 @@ class PyCleo(PythonPackage):
...
 
-    version("1.0.0a5", sha256="097c9d0e0332fd53cc89fc11eb0a6ba0309e6a3933c08f7b38558555486925d3")
+    maintainers("LydDeb")
+
     version(
-        "0.8.1",
-        sha256="3d0e22d30117851b45970b6c14aca4ab0b18b1b53c8af57bed13208147e4069f",
+        "2.0.1",
+        sha256="eb4b2e1f3063c11085cebe489a6e9124163c226575a3c3be69b2e51af4a15ec5",
         preferred=True,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]: Can remove preferred=True, we always default to the latest stable release

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 2]
```diff
@@ -12,17 +12,24 @@ class PyCleo(PythonPackage):
...
+    version("0.8.1", sha256="3d0e22d30117851b45970b6c14aca4ab0b18b1b53c8af57bed13208147e4069f")
 
     depends_on("python@2.7,3.4:3", type=("build", "run"))
     depends_on("python@3.7:3", when="@1:", type=("build", "run"))
     depends_on("py-poetry-core@1:", type="build")
     depends_on("py-poetry-core@1", when="@1:", type="build")
+    depends_on("py-poetry-core@1.1:1", when="@1:2.0.0", type="build")
+    depends_on("py-poetry-core@1.1.0:", when="@2.0.1", type="build")
     depends_on("py-clikit@0.6.0:0.6", when="@0.8.1", type=("build", "run"))
     depends_on("py-pylev@1.3:1", when="@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]: Is this dep still needed?

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 3]
```diff
@@ -12,17 +12,24 @@ class PyCleo(PythonPackage):
...
     depends_on("py-crashtest@0.3.1:0.3", when="@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]: Needs to be updated

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 4]
```diff
@@ -12,17 +12,24 @@ class PyCleo(PythonPackage):
...
+    depends_on("py-rapidfuzz", when="@2:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]:
```suggestion
    depends_on("py-rapidfuzz@2.2", when="@2:", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cleo/package.py (hunk 2 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-cleo/package.py]: In pyproject.toml of py-cleo-0.8.1:
```
[build-system]
requires = ["poetry-core>=1.0.0a5"]
```

In pyproject.toml of py-cleo-1.0.0 and py-cleo-2.0.0:
```
[build-system]
requires = ["poetry-core~=1.1.0"]
```
Which exclude poetry-core@2

In pyproject.toml of py-cleo-2.0.1:
```
[build-system]
requires = ["poetry-core>=1.1.0"]
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-cleo/package.py]: My comment was on pylev, not poetry-core

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 5]
```diff
@@ -12,17 +12,19 @@ class PyCleo(PythonPackage):
     homepage = "https://github.com/sdispater/cleo"
     pypi = "cleo/cleo-0.8.1.tar.gz"
 
+    maintainers("LydDeb")
+
+    version("2.0.1", sha256="eb4b2e1f3063c11085cebe489a6e9124163c226575a3c3be69b2e51af4a15ec5")
+    version("2.0.0", sha256="fbc5cb141cbc31ea8ffd3d5cd67d3b183fa38aa5098fd37e39e9a953a232fda9")
     version("1.0.0a5", sha256="097c9d0e0332fd53cc89fc11eb0a6ba0309e6a3933c08f7b38558555486925d3")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]: Do we want to deprecate the 1.0.0a5 alpha release now that the final release is out?

File: var/spack/repos/builtin/packages/py-cleo/package.py (hunk 2 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-cleo/package.py]: My bad, indeed it is a dependence of version 1.0.0a5.

File: var/spack/repos/builtin/packages/py-cleo/package.py (hunk 5 above)
Comment (LydDeb) [reply var/spack/repos/builtin/packages/py-cleo/package.py]: Good idea, I add version 1.0.0 and deprecate 1.0.0a5.

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 6]
```diff
@@ -12,17 +12,24 @@ class PyCleo(PythonPackage):
...
     depends_on("python@3.7:3", when="@1:", type=("build", "run"))
     depends_on("py-poetry-core@1:", type="build")
-    depends_on("py-poetry-core@1", when="@1:", type="build")
+    depends_on("py-poetry-core@1.1:1", when="@1:2.0.0", type="build")
+    depends_on("py-poetry-core@1.1.0:", when="@2.0.1", type="build")
     depends_on("py-clikit@0.6.0:0.6", when="@0.8.1", type=("build", "run"))
-    depends_on("py-pylev@1.3:1", when="@1:", type=("build", "run"))
-    depends_on("py-crashtest@0.3.1:0.3", when="@1:", type=("build", "run"))
+    depends_on("py-pylev@1.3:1", when="@1.0.0a5", type=("build", "run"))
+    depends_on("py-crashtest@0.3.1:0.3", when="@1.0.0:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]: This is ^0.4.1 now

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 7]
```diff
@@ -12,17 +12,27 @@ class PyCleo(PythonPackage):
...
     depends_on("py-poetry-core@1:", type="build")
-    depends_on("py-poetry-core@1", when="@1:", type="build")
+    depends_on("py-poetry-core@1.1:1", when="@1:2.0.0", type="build")
+    depends_on("py-poetry-core@1.1.0:", when="@2.0.1:", type="build")
     depends_on("py-clikit@0.6.0:0.6", when="@0.8.1", type=("build", "run"))
-    depends_on("py-pylev@1.3:1", when="@1:", type=("build", "run"))
-    depends_on("py-crashtest@0.3.1:0.3", when="@1:", type=("build", "run"))
+    depends_on("py-pylev@1.3:1", when="@1.0.0a5", type=("build", "run"))
+    depends_on("py-crashtest@0.3.1:0.3", when="@1.0.0:", type=("build", "run"))
+    depends_on("py-crashtest@0.4.1:0", when="@2.0.1:2.1.0", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]:
```suggestion
    depends_on("py-crashtest@0.4.1:0", when="@1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 8]
```diff
@@ -12,17 +12,27 @@ class PyCleo(PythonPackage):
...
+    depends_on("py-rapidfuzz@2.2.0:2", when="@1.0.0:2.0", type=("build", "run"))
+    depends_on("py-rapidfuzz@3.0.0:3", when="@2.0.1", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]:
```suggestion
    depends_on("py-rapidfuzz@2.2:2", when="@1:2.0", type=("build", "run"))
    depends_on("py-rapidfuzz@3", when="@2.1:", type=("build", "run"))
```

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 9]
```diff
@@ -12,17 +12,27 @@ class PyCleo(PythonPackage):
...
     )
+    version("0.8.1", sha256="3d0e22d30117851b45970b6c14aca4ab0b18b1b53c8af57bed13208147e4069f")
 
     depends_on("python@2.7,3.4:3", type=("build", "run"))
     depends_on("python@3.7:3", when="@1:", type=("build", "run"))
     depends_on("py-poetry-core@1:", type="build")
-    depends_on("py-poetry-core@1", when="@1:", type="build")
+    depends_on("py-poetry-core@1.1:1", when="@1:2.0.0", type="build")
+    depends_on("py-poetry-core@1.1.0:", when="@2.0.1:", type="build")
     depends_on("py-clikit@0.6.0:0.6", when="@0.8.1", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]:
```suggestion
    depends_on("py-clikit@0.6", when="@0.8.1", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
I'm ignoring 1.0.0a5 because it's deprecated and it makes version ranges confusing because in Spack 1.0.0a5 > 1.0.0 but in pip it's the reverse. I'm also going to remove it after the next Spack release in a couple weeks anyway.

File: var/spack/repos/builtin/packages/py-rapidfuzz/package.py [hunk 10]
```diff
@@ -13,12 +13,14 @@ class PyRapidfuzz(PythonPackage):
     pypi = "rapidfuzz/rapidfuzz-1.8.2.tar.gz"
 
     version("3.3.1", sha256="6783b3852f15ed7567688e2e358757a7b4f38683a915ba5edc6c64f1a3f0b450")
+    version("2.2.0", sha256="acb8839aac452ec61a419fdc8799e8a6e6cd21bed53d04678cdda6fba1247e2f")
     version("1.8.2", sha256="d6efbb2b6b18b3a67d7bdfbcd9bb72732f55736852bbef823bdf210f9e0c6c90")
 
     depends_on("python", type=("build", "link", "run"))
     depends_on("py-setuptools@42:", when="@3:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rapidfuzz/package.py]:
```suggestion
    depends_on("py-setuptools@42:", when="@2:", type="build")
```

File: var/spack/repos/builtin/packages/py-rapidfuzz/package.py [hunk 11]
```diff
@@ -13,12 +13,14 @@ class PyRapidfuzz(PythonPackage):
...
     depends_on("py-setuptools", type="build")
     depends_on("py-scikit-build@0.17", when="@3:", type="build")
+    depends_on("py-scikit-build@0.13:", when="@2.2:", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-rapidfuzz/package.py:23]: Missing build dep:
```
    "rapidfuzz_capi==1.0.5"
```
Missing build/run dep:
```python
    "install_requires": ["jarowinkler >= 1.2.0, < 2.0.0"],                               
```

File: var/spack/repos/builtin/packages/py-cleo/package.py [hunk 12]
```diff
@@ -12,17 +12,26 @@ class PyCleo(PythonPackage):
...
     depends_on("py-poetry-core@1:", type="build")
-    depends_on("py-poetry-core@1", when="@1:", type="build")
-    depends_on("py-clikit@0.6.0:0.6", when="@0.8.1", type=("build", "run"))
-    depends_on("py-pylev@1.3:1", when="@1:", type=("build", "run"))
-    depends_on("py-crashtest@0.3.1:0.3", when="@1:", type=("build", "run"))
+    depends_on("py-poetry-core@1.1:1", when="@1:2.0.0", type="build")
+    depends_on("py-poetry-core@1.1.0:", when="@2.0.1:", type="build")
+    depends_on("py-clikit@0.6", when="@0.8.1", type=("build", "run"))
+    depends_on("py-pylev@1.3:1", when="@1.0.0a5", type=("build", "run"))
+    depends_on("py-crashtest@0.4.1:0", when="@1:", type=("build", "run"))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-cleo/package.py]:
```suggestion
    depends_on("py-crashtest@0.4.1:0.4", when="@1:", type=("build", "run"))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**